pkg/client/config/webpack.common.js

```diff
@@ -169,10 +171,22 @@ module.exports = (env) => {
       ],
     },
     plugins: [
-      new ForkTsCheckerWebpackPlugin({ typescript: { memoryLimit: 4096 } }),
       new HtmlWebpackPlugin({
-        template: path.resolve(__dirname, "../public/index.html"),
-        favicon: path.resolve(__dirname, "../public/favicon.ico"),
+        ...(env !== 'production' 
+          ? {
+              // In dev mode, populate window._env at build time
+              filename: 'index.html',
+              template: path.resolve(__dirname, "../public/index.html.ejs"),
+              favicon: path.resolve(__dirname, "../public/favicon.ico"),
+              templateParameters: {
+                _env: helpers.getEncodedEnv(),
+              },
+            }
+          : {
+              // In real prod mode, populate window._env at run time with express
+              filename: 'index.html.ejs',
+              template: `!!raw-loader!${path.resolve(__dirname, '../public/index.html.ejs')}`,
+            }),
```

Comment:
As we are using dev and prod config files for webpack , wouldn't it make sense to split this respectively ?